src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -205,49 +205,50 @@ string CARMACloudPlugin::updateTags(string str,string tagout, string tagin)
 
 void CARMACloudPlugin::CARMAResponseHandler(QHttpEngine::Socket *socket)
 {
-	QString st; 
+	QByteArray st; 
 	while(socket->bytesAvailable()>0)
 	{	
 		auto readBytes = socket->readAll();
-		if (socket->headers().keys().contains(CONTENT_ENCODING_KEY) && std::string(socket->headers().constFind(CONTENT_ENCODING_KEY).value().data()) == CONTENT_ENCODING_VALUE)
-        {
-			//readBytes is compressed in gzip format
-            st.append(UncompressBytes(readBytes));			
-        }else{
-			st.append(readBytes);
-		}
+		st.append(readBytes);
 	}
-	QByteArray array = st.toLocal8Bit();
-
-	char* _cloudUpdate = array.data(); // would be the cloud update packet, needs parsing
-	
-	
-	string tcm = _cloudUpdate;
 
-	PLOG(logINFO) << "Received TCM from cloud" << tcm << std::endl;
-	if(tcm.length() == 0)
+	if(st.size() == 0)
 	{
-		PLOG(logERROR) << "Received TCM length is zero, and skipped." << std::endl;
+		PLOG(logERROR) << "Received TCM is empty, and skipped." << std::endl;
 		return;
 	}
+	PLOG(logINFO) << "Received TCM bytes size: " << st.size()<< std::endl;
+
+	string tcm = "";
```

Comment:
Just for consistency's sake, is there a reason we use `string` here and `std::string` below?

Reply:
updated